kolibri/plugins/learn/assets/test/views/content-card-group-carousel.spec.js

```diff
@@ -0,0 +1,199 @@
+/* eslint-env mocha */
+import Vue from 'vue-test';
+import { mount as aMount} from 'avoriaz';
+import assert from 'assert';
+import contentCardGroupCarousel from '../../src/views/content-card-group-carousel';
+import makeContentData from '../util/makeContentData';
+// import card from '../../src/views/content-card';
+
+// Always use our vue-test environment
+const mount =
+  (component, options = {}) => aMount(component, Object.assign(options, {instance: Vue}));
+
+// Things to test
+
+// Props from parent
+
+// Data retrieved from store (none)
+
+// User-generated data (none) + events
+
+// lifecycle (none) - but maybe there should be.
+
+// User Stories
+
+// Test structure
+// Set up, render/run, Interact, Assert
+
+// mount with propsData
+
+// What should always be true?
+
+// Conditionals: Are there v-ifs?
+
+// Input
+
+// Happy vs Sad Paths
+
+describe('ContentCardGroupCarousel', function() {
+  describe('Computed Props', function() {
+    beforeEach('mount a carousel', function() {
+      this.wrapper = mount(contentCardGroupCarousel);
+      this.wrapper.setProps({genContentLink: () => '/'});
+    });
+    describe('.contentSetSize()', function() {
+      it('reflects how many cards comfortably fit in the viewport', function() {
+        const wrapper = this.wrapper;
+        const carousel = wrapper.instance();
+        const contents = makeContentData(4);
+
+        wrapper.setProps({contents});
+
+        carousel.elSize.width =  3 * carousel.contentCardWidth;
+        assert.notEqual(carousel.contentSetSize, 3, 'does not account for gutters');
+        assert.equal(carousel.contentSetSize, 2);
+
+        carousel.elSize.width += 2 * carousel.gutterWidth;
+        assert.equal(carousel.contentSetSize, 3);
+
+        carousel.elSize.width += carousel.gutterWidth;
+        assert.equal(carousel.contentSetSize, 3, 'should be 3, not enough room for another card')
+
+      });
+      it('should never be smaller than 1', function() {
+        const wrapper = this.wrapper;
+        const carousel = this.wrapper.instance();
+        const contents = makeContentData(1);
+
+        wrapper.setProps({contents});
+        assert.equal(carousel.contentSetSize, 1, 'is not 1 when contents only has 1 item');
+
+        carousel.elSize.width = 2;
+        assert.equal(carousel.contentSetSize, 1, 'is not 1 when width it tiny');
+
+        carousel.elSize.width = 0;
+        assert.equal(carousel.contentSetSize, 1, 'is not 1 when width is 0');
```

Comment:
multiple `it`s with an error message. Is this weird?

Reply:
I prefer to break these into separate test cases (if you want to do the same setup for each, you can do a `beforeEach` section for common setup code).

The reason I prefer this is that it means that if the first assertion is violated, other assertions will also be tested if they are each in separate tests.